purposes-archiver/src/main/scala/it/pagopa/interop/purposesarchiver/service/impl/QueueServiceImpl.scala

```diff
@@ -0,0 +1,19 @@
+package it.pagopa.interop.purposesarchiver.service.impl
+
+import it.pagopa.interop.purposesarchiver.service.QueueService
+import it.pagopa.interop.commons.queue.message.Message
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import it.pagopa.interop.commons.queue.impl.SQSReader
+import it.pagopa.interop.agreementprocess.events.ArchiveEvent
+import it.pagopa.interop.agreementprocess.events.Events._
+
+final class QueueServiceImpl(queueName: String, visibilityTimeoutInSeconds: Int)(implicit
+  blockingEc: ExecutionContextExecutor
+) extends QueueService {
+
+  val sQSReader: SQSReader = new SQSReader(queueName, visibilityTimeoutInSeconds)(f => _.convertTo[ArchiveEvent])
+
+  override def processMessages(fn: Message => Future[Unit]): Future[Unit] =
+    sQSReader.handleN(10)(fn).map(_ => ())
```

Comment:
SQSReader works on Message type, structurally related to the akka events.
Here must be used SQSHandler.

Reply:
SQSHandler is not useful for receiving messages as a long consumer process, better to use SQSReader . Already spoken with Antonio about this.